src/main/java/vazkii/botania/common/item/lens/LensMagnet.java

```diff
@@ -30,50 +32,59 @@ public void updateBurst(IManaBurst burst, ThrowableEntity entity, ItemStack stac
 		boolean magnetized = entity.getPersistentData().contains(TAG_MAGNETIZED);
 		int range = 3;
 
-		magnetize: {
+		BlockPos source = burst.getBurstSourceBlockPos();
+		final boolean sourceless = source.getY() == -1;
+
+		Predicate<TileEntity> predicate = tile -> tile instanceof IManaReceiver
+				&& (sourceless || tile.getPos().distanceSq(source) > 9)
+				&& ((IManaReceiver) tile).canReceiveManaFromBursts()
+				&& !((IManaReceiver) tile).isFull();
+
+		TileEntity tile = null;
+		if (magnetized) {
+			tile = entity.world.getTileEntity(new BlockPos(
+					entity.getPersistentData().getInt(TAG_MAGNETIZED_X),
+					entity.getPersistentData().getInt(TAG_MAGNETIZED_Y),
+					entity.getPersistentData().getInt(TAG_MAGNETIZED_Z)
+			));
+			if (!predicate.test(tile)) {
+				tile = null;
+				entity.getPersistentData().remove(TAG_MAGNETIZED);
```

Comment:
should `magnetized` also be set to false here so the logic below runs?

Reply:
You're right.